### PR TITLE
Support multiple sinks and multiple processors

### PR DESCRIPTION
**Pull request by kowshikn**
*Issue #47 *

*Description of changes:*

Made the PipelineConfiguration object strict to support. 
* Only one non-null **source**
* Only one optional **buffer**
* Zero or more optional **processors**
* One or more **sinks**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by laneholloway — APPROVED** [review]

File: situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java [hunk 1]
```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
         this.readBatchDelay = getReadBatchDelayFromConfiguration(delay);
     }
 
-    private PluginSetting getSourceFromConfiguration(final Map<String, Map<String, Object>> sourceConfiguration) {
-        final PluginSetting sourcePluginSetting = getFirstPluginSettingFromConfiguration(sourceConfiguration,
-                SOURCE_COMPONENT);
-        if (sourcePluginSetting == null) {
+    private PluginSetting getSourceFromConfiguration(final Map.Entry<String, Map<String, Object>> sourceConfiguration) {
```
Comment (yadavcbala) [inline situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java:44]: You need to sync ur forked master, some of these changes are already in current master

File: situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java [hunk 2]
```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
...
+        }
+        return getPluginSettingFromConfiguration(bufferConfiguration);
     }
 
     private List<PluginSetting> getSinksFromConfiguration(
-            final Map<String, Map<String, Object>> sinkConfiguration) {
-        final List<PluginSetting> sinkPluginSettings = getAllPluginSettingsFromConfiguration(sinkConfiguration);
-        if (sinkPluginSettings.isEmpty()) {
+            final List<Map.Entry<String, Map<String, Object>>> sinkConfigurations) {
+        if (sinkConfigurations==null || sinkConfigurations.isEmpty()) {
```
Comment (yadavcbala) [inline situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java:60]: nit: space - will report in checkstyle and will fail when we enforce them.

`sinkConfigurations == null || sinkConfigurations.isEmpty()`

File: situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java [hunk 3]
```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
...
             throw new IllegalArgumentException("Invalid configuration, at least one sink is required");
         }
-        return sinkPluginSettings;
+        return sinkConfigurations.stream().map(PipelineConfiguration::getPluginSettingFromConfiguration).collect(Collectors.toList());
+    }
+
+    private List<PluginSetting> getProcessorsFromConfiguration(
+            final List<Map.Entry<String, Map<String, Object>>> processorConfigurations) {
+        if (processorConfigurations==null || processorConfigurations.isEmpty()) {
```
Comment (yadavcbala) [inline situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java:68]: same as above

File: situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java [hunk 4]
```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
...
+            return Collections.emptyList();
+        }
+        return processorConfigurations.stream().map(PipelineConfiguration::getPluginSettingFromConfiguration).collect(Collectors.toList());
+    }
+
+
+
+    private static PluginSetting getPluginSettingFromConfiguration(
+            final Map.Entry<String, Map<String, Object>> processorConfiguration) {
```
Comment (yadavcbala) [inline situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java:77]: Nit: rename `processorConfiguration` to `configuration`

File: situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java [hunk 5]
```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
...
-    private PluginSetting getBufferFromConfiguration(final Map<String, Map<String, Object>> bufferConfiguration) {
-        return getFirstPluginSettingFromConfiguration(bufferConfiguration, BUFFER_COMPONENT);
+        return getPluginSettingFromConfiguration(sourceConfiguration);
     }
 
-    private List<PluginSetting> getProcessorsFromConfiguration(
-            final Map<String, Map<String, Object>> processorConfiguration) {
-        return getAllPluginSettingsFromConfiguration(processorConfiguration);
+    private PluginSetting getBufferFromConfiguration(final Map.Entry<String, Map<String, Object>> bufferConfiguration) {
+        if(bufferConfiguration==null) {
```
Comment (yadavcbala) [inline situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java:52]: nit: space pls

**Review by yadavcbala — APPROVED** [review]

**Pull request merged**